src/linux/processor.rs

```diff
@@ -260,6 +261,34 @@ pub fn get_cpu_frequency(cpu_core_index: usize) -> u64 {
         .unwrap_or_default()
 }
 
+pub fn get_physical_core_numbers() -> u64 {
+    let mut s = String::new();
+    if File::open("/proc/cpuinfo")
+        .and_then(|mut f| f.read_to_string(&mut s))
+        .is_err()
+    {
+        return 0;
+    }
+
+    let mut core_ids_and_physical_ids: HashSet<String> = HashSet::new();
+    let mut core_id = "";
+    let mut physical_id = "";
+    for line in s.split('\n') {
+        if line.starts_with("core id") {
+            core_id = line.split(':').last().map(|x| x.trim()).unwrap_or_default();
+        } else if line.starts_with("physical id") {
+            physical_id = line.split(':').last().map(|x| x.trim()).unwrap_or_default();
+        }
+        if !core_id.is_empty() && !physical_id.is_empty() {
+            core_ids_and_physical_ids.insert(format!("{} {}", core_id, physical_id));
+            core_id = "";
+            physical_id = "";
+        }
+    }
+
+    core_ids_and_physical_ids.len() as u64
```

Comment:
```suggestion
   core_ids_and_physical_ids.len() as _
```

Reply:
This change is no longer needed as this function now returns `usize`